Allow Dynamically Priced Parts to be Replaced in Repair Bay

IllianiCBT opened the pull request:
Fix #6515
Fix #6516

### Dev Notes
Basically, some parts have their value dynamically determined based on the tonnage of the unit they're being attached to. In those cases we were comparing sticker price -- basically the base price with no quality modifiers and so on. However, when a part is classified as 'missing' - for example, when it get blown to smithereens, or is scrapped - it has it's price fixed at 0 C-Bills.

CGL ruled that variable priced items could not be fitted to units of tonnage other than the tonnage they were designed for. A 35-ton 'Mek couldn't use a Lance designed for a 40-ton 'Mek, for example, even though the actual weight of the lance is identical for both.

Originally, we got around this issue by comparing the sticker price of each item. Which ultimately then led us to the problem I described above. Missing variable priced parts were completely prevented from being replaced, because those parts were comparing 0 C-Bills to `x` C-Bills (where `x` is whatever the item would be priced at, if it wasn't missing). As the two sticker prices didn't match, those repair couldn't proceed. No matter how many suitable replacement parts the player _actually_ had.

A few hours and one massive headache later I conjured a solution: now if the missing part is a part that has a variable price, we compare the tonnage of the unit the part was built for to the tonnage of the unit the replacement part was built for. If the two match, we're good, replace away. If they don't, then the player cannot use that replacement part to replace the missing part.

Parts that use fixed values are handled identically to how they have been for years. No functionality changes there.

What has changed, though, is that we now explicitly check to see if the 'podded' status of both the original part and the replacement. So that podded items can't replace unpodded items, and so forth. This check should have been included all along, based on the fact that we were testing for it, however that part of the test was a false positive. Now it works.

### Testing
I took @pheonixstorm's campaign and I stripped every unit down to the CT (or equivalent) and then manually repaired every... single... unit... It was a pain in the bum, but a necessary one.

Everything proceeded as expected.

psikomonkie commented on MekHQ/src/mekhq/campaign/parts/equipment/MissingEquipmentPart.java:
```diff
@@ -204,10 +204,57 @@ public boolean isAcceptableReplacement(Part part, boolean refit) {
-        return getType().equals(equipmentPart.getType())
-                && (getTonnage() == equipmentPart.getTonnage())
-                && (getSize() == equipmentPart.getSize())
-                && Objects.equals(newPart.getStickerPrice(), equipmentPart.getStickerPrice());
```
Could/should sticker price still be compared when both aren't 0?

IllianiCBT replied:
That check was only in place to handle what my code handles better:

See https://bg.battletech.com/forums/index.php/topic,14741.msg340122.html#msg340122 for a fuller explanation. It's the same thread as the one in the code comments, just with an updated link.

psikomonkie replied:
Are you sure there are no other equipment considered an equipment part that would need to be differentiated like this? IIRC this is a broad category. If they have a different price I believe they still shouldn't be considered the same item.

IllianiCBT replied:
You’ll have to forgive me, my headache meds hadn’t kicked in - I can certainly add the check back in and then only use this as a fallback in the event we’re dealing with a missing part.

IllianiCBT replied:
I went ahead and made the change. I also took the opportunity to use the new `isVariableCost` method to allow us to catch equipment like this, in the warehouse, so that it will display the tonnage of the item. As, previously, there was no way to know what tonnage something like Environmental Sealing or AES is. Despite tonnage absolutely being important to those parts.

IllianiCBT commented:
- Refactored the attribute comparison method in `EquipmentPart` to only use tonnage comparison as a fallback.
- Refactored equipment part loading from xml to set `unitTonnageMatters` to `true` if it was previously `false` and the part uses variable costing.
- Refactored equipment part initialization the perform the same action as above.
